newIDE/app/src/ResourcesList/index.js

```diff
@@ -77,44 +84,39 @@ export default class ResourcesList extends React.Component<Props, State> {
     this.props.onDeleteResource(resource);
   };
 
-  _scanForNewResources = () => {
+  _scanForNewResources = (extensions: string, createResource: () => gdResource) => {
     const project = this.props.project;
     const resourcesManager = project.getResourcesManager();
-    console.log('Scanning the project folder for new resources...');
     const projectPath = path.dirname(project.getProjectFile());
 
     const getDirectories = (src, callback) => {
-      glob(src + '/**/*.{png,jpg,jpeg,PNG,JPG,JPEG}', callback);
+      glob(src + '/**/*.{' + extensions + '}', callback);
     };
     getDirectories(projectPath, (err, res) => {
       if (err) {
-        console.log('Error loading ', err);
+        console.error('Error loading ', err);
       } else {
         res.forEach(pathFound => {
           const fileName = path.relative(projectPath, pathFound);
           if (!resourcesManager.hasResource(fileName)) {
-            const imageResource = new gd.ImageResource();
-            imageResource.setFile(fileName);
-            imageResource.setName(fileName);
-            resourcesManager.addResource(imageResource);
-            imageResource.delete();
+            createOrUpdateResource(project, createResource(), fileName)
             console.info(`${fileName} added to project.`);
-          }
+          };
         });
       }
       this.forceUpdate();
     });
   };
 
-  _removeAllUnusedImages = () => {
+  _removeUnusedResources = (resourceType: string) => {
```

Comment:
nitpicking: we could even be more precise by using `ResourceKind`:

```
import { type ResourceKind } from './ResourceSource.flow';

_removeUnusedResources = (resourceType: ResourceKind) => {
...
}

```

This will warn you if you have a typo when calling _removeUnusedResources :)

Reply:
I didn't notice that you have added it there initially sorry. Updated now